### ISSUE TITLE
Allowing pattern with vowel mutation in its name in .stignore doesn't work.

### ISSUE THREAD

**Issue by leonschmitz** [MobiusSync/MobiusSync#140]
My `.stignore` looks like this:

```
!/foo
!/bar
!/bäz
*
```

My intention is to ingore all files and directories except the ones I explicitly want on my iPadOS device. In this case `foo`, `bar` and `bäz`. For `foo` and `bar` this works fine across all my devices (MacOS, Windows, iPadOS and Linux, iOS not tested) but `!/bäz` won't sync on my iPad. When I rename to `baz` and change my ignore file accordingly it syncs as expected. So I assume the character `ä` is to blame here.

Versions:
- iPadOS 17.2
- Möbius Sync Pro 1.23.2

**Comment by SimonPickup** [comment]
Likely this is an upstream syncthing issue. Do you have two non-iOS devices you can test on?

**Comment by leonschmitz** [comment]
Yes, I do.

I just tested the same setup on macOS. It works like expected.

macOS 14.5 (23F79)
Syncthing v1.27.2, macOS (64-bit ARM)
Original `.stignore` content:
```
!/Dokumente/Universität Bonn/
*
```

Syncthing ignores every directory and synchronizes everything in `/Dokumente/Universität Bonn`.

I think it also works on Windows 11. If you like I can check that again.

**Comment by SimonPickup** [comment]
I have reproduced this. Upon sone investigating I'm thinking it may be due to NFC vs NFD. Working on it..

**Comment by SimonPickup** [comment]
I have reproduced the problem on macOS. With `.stignore` content:

```
bäd
bad
```
The file `bad` is ignored by `bäd` is not.

I also found a workaround. If you use .stignore` content:

```
ba*d
```
it matches and ignores both files.

This is all on macOS and I suspect it is because filenames are stored in NFD UTF-8. The same applies to iOS.

I will raise an upstream bug later, and a fix too.

**Comment by SimonPickup** [comment]
Upstream issue is https://github.com/syncthing/syncthing/issues/9597 with PR open and general fix agreed.

I will include in MöbiusSync v1.27.9 ahead of Syncthing.

**Comment by SimonPickup** [comment]
Now available on TestFight, please see #172.
